### PR TITLE
710: csv-external is inside a repeat reports error with xml_instance

### LINKED ISSUE #710: Warn when csv-external is inside a repeat
#### Software and hardware versions 
pyxform v1.x.x, Python v

#### Other information 
Things you tried, stack traces, related issues, suggestions on how to fix it...

### PR DESCRIPTION
**Pull request by lindsay-stevens**
Closes #710

#### Why is this the best possible solution? Were any other approaches considered?

Branch based on deps-update / PR 853 branch so only commit d4149102 and later are relevant.

This PR includes three commits that allow for selecting the preferred approach:

1. The minimal fix + tests
2. A minor refactor to tidy up the relevant code in section.py
3. Adding a validation check + error when a `csv-external` or `xml-external` is inside a repeat, as described in the commit message:

> There doesn't seem to be any reason for it to be allowed, but it does seem like the sort of thing that might be useful in future for scoping instances or entities so disallowing it now leaves that use case open (as opposed to warning (for no strong reason) and then potentially requiring forms to be updated later).

#### What are the regression risks?

None / low - fixes a reported bug.

#### Does this change require updates to documentation? If so, please file an issue [here](https://github.com/XLSForm/xlsform.github.io) and include the link below.

Potentially - if `csv-external` and `xml-external` are not allowed in repeats then the docs / XLSForm template could mention that.

#### Before submitting this PR, please make sure you have:
- [x] included test cases for core behavior and edge cases in `tests`
- [x] run `python -m unittest` and verified all tests pass
- [x] run `ruff format pyxform tests` and `ruff check pyxform tests` to lint code
- [x] verified that any code or assets from external sources are properly credited in comments

### CODE REVIEW

File: pyxform/section.py [hunk 1]
```diff
@@ -138,15 +138,15 @@ def xml_instance(self, survey: "Survey", **kwargs):
 
         for child in self.children:
             repeating_template = None
-            if hasattr(child, "flat") and child.get("flat"):
+            if isinstance(child, Section) and child.flat:
```
Comment (lognaturel) [inline pyxform/section.py:141]: Was trying to remember what `flat` was about: https://github.com/XLSForm/pyxform/issues/741

**Review by lognaturel — APPROVED** [review]
I feel like we could just allow the declarations in repeats and hoist them up to the top level but I'm happy with the error message here and it will lead to cleaner XLSForms.
